#57: Explain how to disable TLS in connections with 7.0.x or older (#61)
exasol/exasol-virtual-schema: doc/dialects/exasol.md

Note that you still need to provide a JDBC connection. This is used by the Virtual Schema adapter internally. It is not used for mass data transfer though. And that is where the performance gain comes from.

## Transport Layer Security (TLS)

With version 7.1 Exasol introduced TLS encryption on the database port. Other ports were TLS capable before that already.

### Using TLS Connections

> To establish a TLS connection with an Exasol virtual schema, you must use Exasol Virtual Schema 5.0.3 or later and connect to an Exasol cluster running Exasol 7.1 or later. The reason why you need the Exasol Virtual Schema 5.0.3 or later is that with this version we built-in a JDBC driver that is TLS-capable.

Reviewer: ```suggestion
To establish a TLS connection with an Exasol Virtual Schema, you must use Exasol Virtual Schema 5.0.3 or later and connect to an Exasol cluster running Exasol 7.1 or later. The reason why you need the Exasol Virtual Schema 5.0.3 or later is that with this version we built-in a JDBC driver that is TLS-capable.
```

Reviewer: > The reason why you need the Exasol Virtual Schema 5.0.3 or later is that with this version we built-in a JDBC driver that is TLS-capable.

not sure if that's relevant for users.

Reviewer 2: As far as I can see from change log Exasol JDBC driver dependency was updated in version 5.0.4.

> To establish a TLS connection with an Exasol virtual schema, you must use Exasol Virtual Schema 5.0.3 or later and connect to an Exasol cluster running Exasol 7.1 or later. The reason why you need the Exasol Virtual Schema 5.0.3 or later is that with this version we built-in a JDBC driver that is TLS-capable.

Reviewer 3: ```suggestion
To establish a TLS connection with an Exasol virtual schema, you must use Exasol Virtual Schema 5.0.4 or later and connect to an Exasol cluster running Exasol 7.1 or later. The reason why you need the Exasol Virtual Schema 5.0.4 or later is that with this version we built-in a JDBC driver that is TLS-capable.
```

> To establish a TLS connection with an Exasol Virtual Schema, you must use Exasol Virtual Schema 5.0.3 or later and connect to an Exasol cluster running Exasol 7.1 or later. The reason why you need the Exasol Virtual Schema 5.0.3 or later is that with this version we built-in a JDBC driver that is TLS-capable.

Reviewer 3: ```suggestion
To establish a TLS connection with an Exasol Virtual Schema, you must use Exasol Virtual Schema 5.0.4 or later and connect to an Exasol cluster running Exasol 7.1 or later. The reason why you need the Exasol Virtual Schema 5.0.4 or later is that with this version we built-in a JDBC driver that is TLS-capable.
```

Check the [JDBC driver documentation](https://docs.exasol.com/connect_exasol/drivers/jdbc.htm) in our online handbook for details.

### Disabling TLS

> If you want to connect to a cluster running Exasol 7.0.x or earlier with a recent Exasol Virtual Schema (5.0.3 or later), you _must_ explicitly disable TLS on in the JDBC connection the Virtual Schema uses, otherwise the driver will attempt to establish a TLS connection to the server that does not support it and the connection attempt will fail.

Reviewer 2: As far as I can see from change log Exasol JDBC driver dependency was updated in version 5.0.4.

Reviewer 2: As discussed in PM, "must" part might be to strong.

Reviewer 2: @kaklakariada do you know if my comments will be addressed?

Reviewer 3: @allipatev: I implemented your review findings. Please check if it is OK now.

> If you want to connect to a cluster running Exasol 7.0.x or earlier with a recent Exasol Virtual Schema (5.0.3 or later), you _must_ explicitly disable TLS on in the JDBC connection the Virtual Schema uses, otherwise the driver will attempt to establish a TLS connection to the server that does not support it and the connection attempt will fail.

Reviewer 3: ```suggestion
If you want to connect to a cluster running Exasol 7.0.x or earlier with a recent Exasol Virtual Schema (5.0.4 or later), you _should_ explicitly disable TLS on in the JDBC connection the Virtual Schema uses, otherwise the driver will attempt to establish a TLS connection to the server that does not support it and the connection attempt will fail.
```

The [JDBC driver property `legacyencryption`](https://docs.exasol.com/connect_exasol/drivers/jdbc.htm#SupportedDriverProperties) switches between TLS and the encryption scheme of older Exasol versions. Set it to `1` to disable TLS.

Example:

```sql
CREATE CONNECTION LEGACY_JDBC_CONNECTION
TO 'jdbc:exa:<host>:<port>;legacyencryption=1'
USER '<user>'
IDENTIFIED BY '<password>';
```

> Remember that Virtual Schema adapter (the part rewriting the original query) and the EXALoader (responsible for the `IMPORT` statement execution) can use different drivers! The adapter always uses JDBC, the EXALoader can use EXA connections, an Exasol-proprietary connection variant.

Reviewer: so what does this mean for me as a user? Do I have to adjust mutiple places?
